src/shapes/text.class.js

```diff
@@ -1044,7 +1044,7 @@
           path = this.path,
           shortCut = !isJustify && this.charSpacing === 0 && this.isEmptyStyles(lineIndex) && !path,
           isLtr = this.direction === 'ltr', sign = this.direction === 'ltr' ? 1 : -1,
-          drawingLeft, currentDirection = ctx.canvas.getAttribute('dir');
+          drawingLeft, currentDirection = ctx.direction;
```

Comment:
Can you tell me about this change here?

Reply:
I figured out that everything works correctly on initial loading of the text, but when you change some property like the `textAlign`, the `direction` is not considered anymore (because of the `currentDirection !== this.direction` condition) and the rendering is not correct anymore.

initial loading with `direction=rtl` and `textAlign=rtl`:
<img width="308" alt="Screen Shot 2022-02-14 at 10 46 29 PM" src="https://user-images.githubusercontent.com/26138275/153930964-2d7565b7-1b6c-4a4a-85ac-d7fb2259b7be.png">

after changing the text align to center and then changed it back to left:
<img width="308" alt="Screen Shot 2022-02-14 at 10 47 19 PM" src="https://user-images.githubusercontent.com/26138275/153931057-d88982a0-5039-4c81-b2d8-0917fced81ee.png">

Reply:
i still don't get it.
currentDirection is supposed to mean `direction on which the canvas is set to` and both the attribute and the ctx.direction are kept in sync. How one is different from the other?

Reply:
I logged their values and It seems that the `ctx.direction` is not in sync with the `ctx.canvas.dir`

<img width="284" alt="Screen Shot 2022-02-24 at 10 58 43 PM" src="https://user-images.githubusercontent.com/26138275/155593722-9e48cf59-0a0a-4c55-9180-f7e2cafee4e2.png">

Reply:
oh right, because ctx.direction is involved probably in save and restore while the canvas attribute isn't.
That is right and is a bug on our end

Reply:
@asturur by the way, it seems that the `ctx.direction` is not supported in Firefox as stated [here](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/direction), so this is not

Reply:
yep firefox is a bit rough and is also the reason why i added canvas.attribute.dir.
But eventually we can do browser sniffing and change.